cli/state.rs

```diff
@@ -233,14 +269,50 @@ impl ThreadSafeState {
       _ => None,
     };
 
+    let dir =
+      deno_dir::DenoDir::new(custom_root, &config, progress.clone()).unwrap();
+
+    let main_module: Option<String> = if argv_rest.len() <= 1 {
+      None
+    } else {
+      let specifier = argv_rest[1].clone();
+      let referrer = ".";
+      // TODO: does this really have to be resolved by DenoDir?
+      //  Maybe we can call `resolve_module_spec`
+      match dir.resolve_module_url(&specifier, referrer) {
```

Comment:
Our specifier resolving is all over the place:
- modules
- DenoDir
- ops

I propose to introduce struct `Specifier`/`ModuleSpecifier` and pass it around instead of raw strings

Reply:
Let's do that in followup PR

Reply:
Actually there are a few functions that take (specifier, referrer) args which I would like to replace with a single absolute URL arg.

I think we need to have a special “resolve” op and then only use URLs.

(On phone - I’ll copy some links)